test/libsolidity/SemanticTest.cpp

```diff
@@ -46,77 +46,102 @@ SemanticTest::SemanticTest(string const& _filename, string const& _ipcPath, lang
 	m_source = parseSourceAndSettings(file);
 	if (m_settings.count("compileViaYul"))
 	{
-		m_validatedSettings["compileViaYul"] = m_settings["compileViaYul"];
-		m_compileViaYul = true;
-		m_settings.erase("compileViaYul");
+		if (m_settings["compileViaYul"] == "also")
+		{
+			m_validatedSettings["compileViaYul"] = m_settings["compileViaYul"];
+			m_runWithYul = true;
+			m_runWithoutYul = true;
+		}
+		else
+		{
+			m_validatedSettings["compileViaYul"] = "only";
+			m_runWithYul = true;
+			m_runWithoutYul = false;
+		}
 	}
 	parseExpectations(file);
 }
 
 TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePrefix, bool _formatted)
 {
+	for(bool compileViaYul: set<bool>{!m_runWithoutYul, m_runWithYul})
```

Comment:
wouldn't this run ALL tests twice?

Reply:
Cases:
- ``m_runWithoutYul = false, m_runWithYul = false``:
   cannot happen
- ``m_runWithoutYul = true, m_runWithYul = false``:
  ``set_<bool>{!m_runWithoutYul, m_runWithYul}  = {false, false} = {false}``
   => runs only with ``compileViaYul = false``
- ``m_runWithoutYul = false, m_runWithYul = true``:
  ``set_<bool>{!m_runWithoutYul, m_runWithYul}  = {true, true} = {true}``
   => runs only with ``compileViaYul = true``
- ``m_runWithoutYul = true, m_runWithYul = true``:
  ``set_<bool>{!m_runWithoutYul, m_runWithYul}  = {false,true} ``
   => runs with ``compileViaYul = false`` and with ``compileViaYul = true``

Reply:
Or if it's not the values you were concerned about, but their multiplicity: it's a ``std::set`` - it'll never contain the same value twice.

Reply:
Right, i forgot that (main) aspect of a set. Not used seeing it in this context

Reply:
I think it's in fact the shortest way to express this :-). But let me know if you come up with something even shorter!